app/controllers/IndividualsDetailsController.scala

```diff
@@ -45,19 +46,34 @@ class IndividualsDetailsController  @Inject()(authConnector: AuthConnector,
             logger.warn(s"User with NINO ${authContext.nino.value} is trying to access NINO $nino")
             Future(Results.Unauthorized("You are not authorised to access this resource"))
           }else {
-            individualDetailsService.getIndividualDetails(nino, resolveMerge).map(resultFromStatus)
+            individualDetailsService.getIndividualDetails(nino, resolveMerge).map { response =>
+              val headers = response.headers
+              val correlationIdHeader:String = headers.find(_._1 == "CorrelationId") match {
+                case Some((_, value)) => value(0)
+                case None =>
+                  logger.warn("CorrelationId not found in the response headers")
+                  ""
+              }
+              resultFromStatus(response, correlationIdHeader)
+            }
           }
         }
       }
   }
 
-  private def resultFromStatus(response: HttpResponse): Result = {
+  private def resultFromStatus(response: HttpResponse, correlationId: String): Result = {
     response.status match {
       case OK => Results.Ok(response.body)
       case BAD_REQUEST => Results.BadRequest(response.body)
       case UNAUTHORIZED => Results.Unauthorized(response.body)
-      case NOT_FOUND => Results.NotFound(response.body)
-      case INTERNAL_SERVER_ERROR => Results.InternalServerError(response.body)
```

Comment:
Why only certain cases? Can't we just log the correlation id when we generate it?